### PR TITLE
move testing to vitest

### PR DESCRIPTION
**Pull request by bmschmidt**
Still failing a test on wait_for_project_lock.

----

> [!IMPORTANT]
> Migrate testing framework from `uvu` to `vitest`, updating test scripts, dependencies, and test files accordingly.
> 
>   - **Testing Framework Migration**:
>     - Replace `uvu` with `vitest` in `package.json` test scripts.
>     - Remove `uvu` and related dependencies; add `vitest`.
>     - Add `vitest.config.ts` for configuration.
>   - **Test Files**:
>     - Convert test files from `.js` to `.ts` (e.g., `arrow.test.js` to `arrow.ts`).
>     - Update test syntax from `uvu` to `vitest` in `embedding.test.ts`, `neighbors.test.ts`, `project.test.ts`, and `user.test.ts`.
>     - Replace `assert` with `expect` in test assertions.
>   - **Miscellaneous**:
>     - Remove `project.test.js` and replace with `project.test.ts` with updated test cases.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=nomic-ai%2Fts-nomic&utm_source=github&utm_medium=referral)<sup> for ecd2422e4a283ed69ee184ff0a8278fd6555dd41. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by bmschmidt** [comment]
* **#83** <a href="https://app.graphite.dev/github/pr/nomic-ai/ts-nomic/83?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bmschmidt and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tests/arrow.ts
Comment (RLesser) [inline tests/arrow.ts:1]: no new import?

**Review by RLesser — APPROVED** [review]
LGTM, address human and bot comments

should we add a CI check for test passing? and isn't there a `tsc` check as well?

**Comment by bmschmidt** [comment]
### Merge activity

* **Nov 4, 10:02 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/nomic-ai/ts-nomic/83).

**Pull request merged**